crates/chain/src/tx_graph.rs

```diff
@@ -449,6 +449,32 @@ impl<A> TxGraph<A> {
     pub fn is_empty(&self) -> bool {
         self.txs.is_empty()
     }
+
+    /// Map TxGraph for another anchor type
+    pub fn map_anchors<A2: Ord, F>(self, f: F) -> TxGraph<A2>
+    where
+        F: Fn(A) -> A2,
+    {
+        TxGraph::<A2> {
+            txs: HashMap::from_iter(self.txs.into_iter().map(
+                |(txid, (tx_node_interal, set, amt))| {
+                    (
+                        txid,
+                        (
+                            tx_node_interal.clone(),
+                            BTreeSet::from_iter(set.into_iter().map(&f)),
+                            amt,
+                        ),
+                    )
+                },
+            )),
+            anchors: BTreeSet::<(A2, Txid)>::from_iter(
+                self.anchors.into_iter().map(|(a, txid)| (f(a), txid)),
+            ),
```

Comment:
This is written in a way which assumes the caller's implementation of `F` is deterministic. A.K.A passing in the same `A` on separate calls will give us the same `A2` result.

The benefit of NOT having this assumption is:
1. It's safer.
2. It's probably more performant (as there will be no duplicate calls on `F`).

Reply:
We should write a test to make sure the resultant `TxGraph` is reconstructible.

1. Call `.map_anchor` on `graph_a` with a very strange `F` that is non-deterministic (let's call the resultant `TxGraph` - `graph_b`).
2. Call `graph_b.initial_changeset`. Construct a `graph_c` with the resultant changeset.
3. `assert_eq!(graph_b, graph_c)`.

Also another test that shows `graph_a` and `graph_b` have the same transactions, and last_seens. Also the same anchor count.

Reply:
Got it.  It now map only once on ChangeSet.